apis/python/src/tiledbsoma/util.py

```diff
@@ -91,3 +92,28 @@ def uri_joinpath(base: str, path: str) -> str:
                 parts[2] = parts[2] + "/" + path
 
     return urllib.parse.urlunparse(parts)
+
+
+# TODO: handle slices like `1:` or `:2`
+def ids_to_list(ids: Union[slice, List[int], pa.Array]) -> pa.ChunkedArray:
+    """
+    For the interface between ``SOMADataFrame::read`` et al. (Python) and ``SOMAReader`` (C++): the
+    ``ids`` argument to the former can be slice or list; the argument to
+    ``SOMAReader::set_dim_points`` must be a list.
```

Comment:
When setting a slice for a `SOMAReader` query, we should use `SOMAReader::set_dim_ranges` instead of converting the slice to a list.

This test shows an example:
https://github.com/single-cell-data/TileDB-SOMA/blob/main/libtiledbsoma/test/test_soma_reader.py#L82